src/Meshes/linesMesh.ts

```diff
@@ -155,15 +155,15 @@ export class LinesMesh extends Mesh {
     /**
      * @hidden
      */
-    public get material(): Material {
+    public get material(): ShaderMaterial {

```

Comment:
I will not force it to be ShaderMaterial actually. We could want to use NME for instance

Reply:
No probs. Wasn't sure what impact it might have. Better safe than sorry but happy to change for both getter & setter.

@deltakosh David did you guys have a chance to look at those Azure errors. I believe those have little to do with actual PR change, unless I am missing something.

Reply:
Yes I just did fix them :) It was because of TS being TS. They break back compat on every tiny release
It is fixed now

I will merge your PR after your change

Thanks again for your help btw :)

Reply:
Hey no probs! Re your earlier comment... color shader is of `ShaderMaterial` type
```
private _colorShader: ShaderMaterial;
```
which seems fine looking at operations on shader defines. 
I believe it is better to leave `ShaderMaterial` in getter/setter until NME (what is it?) use case. 

One more small commit coming in.

Reply:
Well my comment is if we open it we need to use Material because you can use NME this way

Reply:
@FullStackForger You should be able to switch to `Material` for the base class of `_colorShader` without too much hassle by performing those changes:

```typescript
private _isShaderMaterial(shader: Material): shader is ShaderMaterial {
    return shader.getClassName() === "ShaderMaterial";
}

private _addClipPlaneDefine(label: string) {
    if (!this._isShaderMaterial(this._colorShader)) {
        return;
    }
    ...
}

private _removeClipPlaneDefine(label: string) {
    if (!this._isShaderMaterial(this._colorShader)) {
        return;
    }
    ...
}

public _bind(subMesh: SubMesh, effect: Effect, fillMode: number): Mesh {
    ...
    if (!this.useVertexColor && this._isShaderMaterial(this._colorShader)) {
        ...
    }
    ...
}
```

Reply:
👌🏻